src/components/ShareBlock/ShareModelBlock.tsx

```diff
@@ -1,64 +1,62 @@
 import React, { useState } from 'react';
 import ShareBlock from './ShareBlock';
-import { STATES } from 'common';
+import { County, Region, State } from 'common/regions';
 
 import EmbedPreview from './EmbedPreview';
-import { County } from 'common/locations';
 import { Projections } from 'common/models/Projections';
+import { findCountyByFips } from 'common/locations';
+import { fail } from 'common/utils';
 
 const BASE_SHARE_URL = 'https://covidactnow.org/us';
 
-interface ShareModelBlockProps {
-  stateId: string;
-  county: County | undefined;
-  projections: Projections;
-  stats: any;
+interface ShareModelBlockParams {
+  region?: Region;
+  projections?: Projections;
+  stats?: any;
 }
 
-const ShareModelBlock: React.FC<ShareModelBlockProps> = ({
-  stateId,
-  county,
+const ShareModelBlock = ({
+  region,
   projections,
   stats,
-}) => {
-  const { displayName, shareURL } = getUrlAndShareQuote(stateId, county);
-  const countyName = county && county.county;
+}: ShareModelBlockParams) => {
+  const { displayName, shareURL } = getUrlAndShareQuote(region);
   const shareQuote = `I'm keeping track of ${displayName}'s COVID data and risk level with @CovidActNow. What does your community look like?`;
   const [showEmbedPreviewModal, setShowEmbedPreviewModal] = useState(false);
-
+  const county = region && findCountyByFips(region.fipsCode);
   return (
     <>
       <ShareBlock
-        stateId={stateId}
+        region={region}
         shareURL={shareURL}
-        countyName={countyName}
         shareQuote={shareQuote}
         projections={projections}
         onClickEmbed={() => setShowEmbedPreviewModal(true)}
         stats={stats}
       />
+      {/* todo: add region to this. county={county} */}
       <EmbedPreview
         open={showEmbedPreviewModal}
-        county={county}
         onClose={() => setShowEmbedPreviewModal(false)}
+        county={county}
```

Comment:
The county here is supposed to be a "location" county, not a region, this will probably fail